compiler/plc_xml/src/xml_parser/tests.rs

```diff
@@ -333,3 +366,180 @@ const ASSIGNMENT_TO_UNRESOLVED_REFERENCE: &str = r#"
+                        <relPosition x="80" y="10"/>
+                    </connectionPointOut>
+                    <expression>10</expression>
+                </inVariable>
+                <outVariable localId="4" height="20" width="80" executionOrderId="1" negated="false" storage="none">
+                    <position x="330" y="100"/>
+                    <connectionPointIn>
+                        <relPosition x="0" y="10"/>
+                        <connection refLocalId="3"/>
+                    </connectionPointIn>
+                    <expression>val</expression>
+                </outVariable>
+                <inOutVariable localId="5" height="20" width="80" negatedIn="false" storageIn="none" negatedOut="false">
+                    <position x="780" y="60"/>
+                    <connectionPointIn>
+                        <relPosition x="0" y="10"/>
+                    </connectionPointIn>
+                    <connectionPointOut>
+                        <relPosition x="80" y="10"/>
+                    </connectionPointOut>
+                    <expression>a</expression>
+                </inOutVariable>
+            </FBD>
+        </body>
+    </pou>
+    "#;
+
+    pub(super) const CONDITIONAL_RETURN_CHAINED_TO_ANOTHER_CONDITIONAL_RETURN: &str = r#"
+    <?xml version="1.0" encoding="UTF-8"?>
+    <pou xmlns="http://www.plcopen.org/xml/tc6_0201" name="conditional_return" pouType="functionBlock">
+        <interface>
+            <localVars/>
+            <addData>
+                <data name="www.bachmann.at/plc/plcopenxml" handleUnknown="implementation">
+                    <textDeclaration>
+                        <content>
+    FUNCTION_BLOCK conditional_return
+    VAR_INPUT
+        val : DINT;
+    END_VAR</content>
+                    </textDeclaration>
+                </data>
+            </addData>
+        </interface>
+        <body>
+            <FBD>
+                <return localId="1" height="20" width="76" executionOrderId="0">
+                    <position x="330" y="60"/>
+                    <addData>
+                        <data name="www.bachmann.at/plc/plcopenxml" handleUnknown="implementation">
+                            <negated value="false"/>
+                        </data>
+                    </addData>
+                </return>
+                <return localId="2" height="20" width="76" executionOrderId="1">
+                    <position x="330" y="60"/>
+                    <connectionPointIn>
+                        <relPosition x="0" y="10"/>
+                        <connection refLocalId="1"/>
+                    </connectionPointIn>
+                    <addData>
+                        <data name="www.bachmann.at/plc/plcopenxml" handleUnknown="implementation">
+                            <negated value="false"/>
+                        </data>
+                    </addData>
+                </return>
+            </FBD>
+        </body>
+    </pou>
+    "#;
+}
```

Comment:
FYI I'm planning on changing the Serializer API such that these lines can be shrinked quite a bit, e.g.
```
Pou::new(name, PouType::Function, content)
    .with_body(vec![
        InVariable::new(localid, negated, Option<expression>, connection),
        Return::new(localid, negated, Option<expression>, connection),
        // ...
    ])
);
```
This will happen in a new PR though, so for now this file just happens to be bloated.